kernel/kernel-api/src/main/java/io/delta/kernel/internal/commit/DefaultFileSystemManagedTableOnlyCommitter.java

```diff
@@ -39,7 +52,37 @@ public CommitResponse commit(
       throws CommitFailedException {
     commitMetadata.getReadProtocolOpt().ifPresent(this::validateProtocol);
     commitMetadata.getNewProtocolOpt().ifPresent(this::validateProtocol);
-    throw new UnsupportedOperationException("Default Committer not yet implemented");
+
+    final String jsonCommitFile =
+        FileNames.deltaFile(commitMetadata.getDeltaLogDirPath(), commitMetadata.getVersion());
+
+    logger.info("Attempting to commit {}", jsonCommitFile);
+
+    try {
+      wrapEngineExceptionThrowsIO(
+          () -> {
+            engine
+                .getJsonHandler()
+                .writeJsonFileAtomically(jsonCommitFile, finalizedActions, false /* overwrite */);
+            return null;
+          },
+          String.format("Write file actions to JSON log file `%s`", jsonCommitFile));
+    } catch (FileAlreadyExistsException e) {
+      throw new CommitFailedException(
+          true /* retryable */,
+          true /* conflict */,
+          "Concurrent write detected for version " + commitMetadata.getVersion(),
+          e);
+    } catch (IOException e) {
```

Comment:
For the retrying: we are mainly undecided about how Kernel / in the future the Connector reacts to failed commit attempts right? There's no debate about what exceptions the committer should throw like here? i.e. is there any reason we should make this isRetryable=false ?

Reply:
Great question. If things are reasonably retryable, we should throw a retryable exception.

Worst case, we try, and the file *already does exist*. Then, really, the question/semantic is what do we do with a `retryable=true, isConflict=true` exception *after* we have already seen a `retryable=true, isConflict=false` exception.

Well, we need to see if that 2nd exception is actually just us seeing the successful write that, somehow (network hiccup) causes the 1st exception!

So Kernel txn.commit rety loop can jsut handle this. You can either throw (totally valid) or try and look at the `.json` file and see which case it is

Reply:
We will document this well inside of txn.commit